src/ecosystem/corals/growth.jl

```diff
@@ -247,6 +272,194 @@ function bleaching_mortality!(Y::AbstractArray{Float64,2},
+   Coral Reefs 37, 527-532.
+   https://doi.org/10.1007/s00338-018-1677-y
+
+4. Baird, A., Madin, J., Álvarez-Noriega, M., Fontoura, L., Kerry, J., Kuo, C.,
+     Precoda, K., Torres-Pulliza, D., Woods, R., Zawada, K., & Hughes, T. (2018).
+   A decline in bleaching suggests that depth can provide a refuge from global
+     warming in most coral taxa.
+   Marine Ecology Progress Series, 603, 257-264.
+   https://doi.org/10.3354/meps12732
+"""
+function bleaching_mortality!(cover::Matrix{Float64}, dhw::Vector{Float64},
+    depth_coeff::Vector{Float64}, dist::Matrix{Truncated}, dist_t1::Matrix{Truncated},
+    prop_mort::SubArray{Float64})::Nothing
+    n_sp_sc, n_locs = size(cover)
+
+    # Adjust distributions for all locations, ignoring juveniles
+    # we assume the high background mortality of juveniles
+    @floop for (sp_sc, loc) in Iterators.product(3:n_sp_sc, 1:n_locs)
+        # Skip if location experiences no heat stress or there is no population
+        if dhw[loc] == 0.0 || cover[sp_sc, loc] == 0.0
+            continue
+        end
+
+        affected_pop::Float64 = cdf(dist[sp_sc, loc], dhw[loc])
+        mort_pop::Float64 = 0.0
+        if affected_pop > 0.0
+            # Calculate depth-adjusted bleaching mortality
+            mort_pop = affected_pop * depth_coeff[loc]
+
+            # Set values close to 0.0 (e.g., 1e-214) to 0.0
+            # https://github.com/JuliaLang/julia/issues/23376#issuecomment-324649815
+            if (mort_pop + one(mort_pop)) ≈ one(mort_pop)
+                mort_pop = 0.0
+            end
+        end
+
+        prop_mort[sp_sc, loc] = mort_pop
+        if mort_pop > 0.0
+            # Re-create distribution
+            d = censored(dist[sp_sc, loc]; lower=mort_pop)
+            μ::Float64 = mean(d)
+            dist_t1[sp_sc, loc] = TruncatedNormal(μ, std(d), 0.0, μ + HEAT_UB)
+
+            # Update population
+            cover[sp_sc, loc] = cover[sp_sc, loc] * (1.0 - mort_pop)
+        end
+    end
+end
+
+"""
+    _merge_distributions!(c_t, c_t1, dists, dists_t1, c_increase)
+
+Combine distributions using the weighted average approach for all size classes above 1.
+If a positive change in cover (between \$t\$ and \$t+1\$) is found for a given size
+class, the distribution of the previous size class is weighted by the difference, and the
+distribution of the current size class is weighted by [current cover - difference].
+Where a negative change has occurred, it is assumed mortalities overcame growth.
+
+# Arguments
+- `c_t` : Cover for given size class, location at timestep \$t\$
+- `c_t1` : Cover for given size class, location at timestep \$t+1\$
+- `dists` : Critical DHW threshold distribution for timestep \$t\$
+- `dists_t1` : Critical DHW threshold distribution for timestep \$t+1\$
+- `c_increase` : Cache matrix to temporarily store difference between \$c_t\$ and \$c_t1\$
+
+# Returns
+Nothing
+"""
+function _merge_distributions!(c_t, c_t1, dists, dists_t1, c_increase)::Nothing
```

Comment:
Very finicky, but is it more consistent to call `dists` 'dists_t` here?

Reply:
Done